packages/jupyter-chat/src/components/input/toolbar-registry.tsx

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (c) Jupyter Development Team.
+ * Distributed under the terms of the Modified BSD License.
+ */
+import * as React from 'react';
+
+import { AttachButton, CancelButton, SendButton } from './buttons';
+import { IInputModel } from '../../input-model';
+import { ISignal, Signal } from '@lumino/signaling';
+
+/**
+ * The toolbar registry interface.
+ */
+export interface IInputToolbarRegistry {
+  /**
+   * A signal emitting when the items has changed.
+   */
+  readonly itemsChanged: ISignal<IInputToolbarRegistry, void>;
+  /**
+   * Get a toolbar item.
+   */
+  get(name: string): InputToolbarRegistry.IToolbarItem | undefined;
+
+  /**
+   * Get the list of the visible toolbar items in order.
+   */
+  getItems(): InputToolbarRegistry.IToolbarItem[];
+
+  /**
+   * Add a toolbar item.
+   */
+  addItem(name: string, item: InputToolbarRegistry.IToolbarItem): void;
+
+  /**
+   * Remove a toolbar item.
+   */
+  removeItem(name: string): void;
```

Comment:
If this method is not used yet, maybe we can remove it from the interface, and add it later if needed?